app/forms/reactivate_profile_form.rb

```diff
@@ -9,9 +9,11 @@ class ReactivateProfileForm
   attr_accessor :recovery_code, :password
   attr_reader :user
 
-  def initialize(user, attrs = {})
+  def initialize(user, attrs = { recovery_code: [] })
```

Comment:
this is not a destructuring assignment like it is in JS, the reason `recovery_code` works is because it's after the `super` call and it's using the `attr_accessor`-defined method. I would just leave this as `attrs = {}`